### PR TITLE
WIP - surface.hpp

### PR DESCRIPTION
**Pull request by pressel**
This PR will add a Surface struct that will be used to move surface relevant data between an aerosol model and host model.

### CODE REVIEW

File: haero/surface.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  // these are supported for initializing containers of Surface
+  Surface(const Surface &rhs) = default;
+  Surface &operator=(const Surface &rhs) = default;
+
+  /// destructor, valid on both host and device
+  KOKKOS_FUNCTION
+  ~Surface() {}
+
+  // land fraction [unitless]
+  Real landfrac = 0.0;
```
Comment (jeff-cohere) [inline haero/surface.hpp:29]: I think the initializations can be omitted here in favor of a default constructor that zeros all the fields.

File: haero/CMakeLists.txt [hunk 2]
```diff
@@ -63,5 +64,6 @@ install(FILES aero_process.hpp
               testing.hpp
               utils.hpp
               root_finders.hpp
+
```
Comment (jeff-cohere) [inline haero/CMakeLists.txt:67]: Is this blank line intentional?

File: haero/surface.hpp (hunk 1 above)
Comment (pressel) [reply haero/surface.hpp]: I removed these initializations, in favor of  `Surface s{};` where relevant in the testing.

File: haero/CMakeLists.txt (hunk 2 above)
Comment (pressel) [reply haero/CMakeLists.txt:67]: It is not.

**Review by jeff-cohere — APPROVED** [review]
This looks good. Let's wait on merging till we have a matching PR for mam4xx.

**Pull request merged**